consensus/bor/span_store.go

```diff
@@ -85,6 +92,45 @@ func (s *SpanStore) getLatestSpan(ctx context.Context) (*borTypes.Span, error) {
 	return s.latestSpanCache.Load(), nil
 }
 
+func (s *SpanStore) updateHeimdallStatus(ctx context.Context) (err error) {
+	var syncInfo *ctypes.SyncInfo
+	if s.heimdallClient == nil {
```

Comment:
This check looks redundant since we only call this function when `s.heimdallClient != nil` ?